src/v/compression/internal/snappy_java_compressor.cc

```diff
@@ -83,9 +76,14 @@ iobuf snappy_java_compressor::uncompress(const iobuf& x) {
     if (unlikely(snappy_magic::java_magic != magic_compare)) {
         return snappy_standard_compressor::uncompress(x);
     }
-    // NOTE: version and min_version are LITTLE_ENDIAN!
-    const auto version = iter.consume_type<int32_t>();
-    const auto min_version = iter.consume_type<int32_t>();
+    // NOTE: version and min_version are BIG_ENDIAN!
+    // Previously, these version fields were erroneously written with
+    // little-endian encoding. They are now corrected to be written and decoded
+    // using big-endian, but we must retain backwards compatibility here with
+    // the existing, improperly encoded batches (as version, min_version fields
+    // with value 1 will decode to the value 16777216).
+    const auto version = iter.consume_be_type<int32_t>();
+    const auto min_version = iter.consume_be_type<int32_t>();
     if (unlikely(min_version < snappy_magic::min_compatible_version)) {
```

Comment:
How does this check not fire before this PR? Do we never decode Snappy values (in tests or in real life)

Reply:
Well as long as min_version >= 1 this check will always pass. I'm pretty sure this check had a good intention but it was implemented incorrectly. The only check that makes sense to me here is `if (snappy_magic::version < min_version) throw ...;`, i.e. allow encoders to require minimum version of a decoder. Probably too late for that...

FWIW, this code is also incorrectly translated from Java. The java check is `version < snappy_magic::min_compatible_version`. min_version is not used by java anywhere directly: https://github.com/search?q=repo%3Axerial/snappy-java%20compatibleVersion&type=code